src/shapes/group.class.js

```diff
@@ -15,280 +21,318 @@
+     * Once an object is selected, instance is rendered without the selected object.
+     * This way instance is cached only once for the entire interaction with the selected object.
+     * @private
      */
-    useSetOnGroup: false,
+    _activeObjects: undefined,
 
     /**
      * Constructor
-     * @param {Object} objects Group objects
+     *
+     * @param {fabric.Object[]} [objects] instance objects
      * @param {Object} [options] Options object
-     * @param {Boolean} [isAlreadyGrouped] if true, objects have been grouped already.
-     * @return {Object} thisArg
-     */
-    initialize: function(objects, options, isAlreadyGrouped) {
-      options = options || {};
-      this._objects = [];
-      // if objects enclosed in a group have been grouped already,
-      // we cannot change properties of objects.
-      // Thus we need to set options to group without objects,
-      isAlreadyGrouped && this.callSuper('initialize', options);
+     * @param {boolean} [objectsRelativeToGroup] true if objects exist in group coordinate plane
+     * @return {fabric.Group} thisArg
+     */
+    initialize: function (objects, options, objectsRelativeToGroup) {
       this._objects = objects || [];
-      for (var i = this._objects.length; i--; ) {
-        this._objects[i].group = this;
-      }
-
-      if (!isAlreadyGrouped) {
-        var center = options && options.centerPoint;
-        // we want to set origins before calculating the bounding box.
-        // so that the topleft can be set with that in mind.
-        // if specific top and left are passed, are overwritten later
-        // with the callSuper('initialize', options)
-        if (options.originX !== undefined) {
-          this.originX = options.originX;
-        }
-        if (options.originY !== undefined) {
-          this.originY = options.originY;
-        }
-        // if coming from svg i do not want to calc bounds.
-        // i assume width and height are passed along options
-        center || this._calcBounds();
-        this._updateObjectsCoords(center);
-        delete options.centerPoint;
-        this.callSuper('initialize', options);
+      this._activeObjects = [];
+      this.__objectMonitor = this.__objectMonitor.bind(this);
+      this.__objectSelectionTracker = this.__objectSelectionMonitor.bind(this, true);
+      this.__objectSelectionDisposer = this.__objectSelectionMonitor.bind(this, false);
+      this.callSuper('initialize', options);
+      if (objectsRelativeToGroup) {
+        this.forEachObject(function (object) {
+          this.enterGroup(object, false);
+        }, this);
       }
       else {
-        this._updateObjectsACoords();
+        //  we need to preserve object's center point in relation to canvas and apply group's transform to it
+        var inv = invertTransform(this.calcTransformMatrix());
+        this.forEachObject(function (object) {
+          var center = transformPoint(object.getCenterPoint(), inv);
```

Comment:
this line was:
```
var center = transformPoint(this.getCenterPoint(), t);
```

The center point was the one from the group, and the matrix was (inv * objectTransform)

now, if i m not wrong the objectTransform would move the the group center ( the 0,0 of this plane ) in the object center, taking in account also the object trasnform. But getCenterPoint is doing the same somehow.

I not 100% this change is equivalent, but fixes the svg import issue. Do you think is breaking something else?

Reply:
is this supposed to take in account creating group with objects that are part of other groups and already deeply transformed?

Reply:
I think this was a bug.
It should be the object center that is transformed into the group's plane.
Great work!

Reply:
I have an idea for a test to write for this. Can I commit to v6!? or should I open a PR?

Reply:
I want to commit this fix to v6! as well.

Reply:
we need to polish and merge this PR, either by you with a patch or we just merge this.
Changing this piece of code break a single UTs, it has some differences for nested groups.

Reply:
I'll patch and merge.
What are the diffs in tests?

Reply:
master:
![image](https://user-images.githubusercontent.com/34343793/160274911-97ee19a1-9ab0-4e9c-b475-a990682ebd29.png)

before your fix:
![image](https://user-images.githubusercontent.com/34343793/160274998-f9d2e888-9ee9-49a2-ae94-3f8e5aa7f975.png)


with your fix:
![image](https://user-images.githubusercontent.com/34343793/160274966-1cf10dd7-174e-4943-a83d-66de63998302.png)

I'll push a visual test for this

Reply:
This test fails because of `group3`
Something is making it spread over the blue rect.